src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheck.java

```diff
@@ -105,17 +128,39 @@
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST parent = ast.getParent();
-        switch (parent.getType()) {
-            case TokenTypes.NOT_EQUAL:
-            case TokenTypes.EQUAL:
-            case TokenTypes.LNOT:
-            case TokenTypes.LOR:
-            case TokenTypes.LAND:
-                log(parent, MSG_KEY);
-                break;
-            default:
-                break;
+        if (TokenUtil.isOfType(parent, UNACCEPTABLE_PARENT_OF_TRUE_OR_FALSE)
```

Comment:
to be honest, previous way is more readable. It clearly shows how tokens are handled. You can just add one more case to switch with if condition. It will be clearer imo

Reply:
done